### ISSUE TITLE
Publish Swagger Definition of the API

### ISSUE THREAD

**Issue by Mohammed90** [monicahq/monica#2650]
Currently the API documentation is only available as a set of webpages that can be [here](https://www.monicahq.com/api). A user who would like to build a client of the API needs to browse 27 different pages and manually collect the information needed to build the client. Publishing Swagger document makes it easier for developers to integrate Monica into their apps and the other way around. It will also make it easy to generate the API client for the mobile app once it's decided to be rebuilt.

P.S.: I tried a bit with [DarkaOnLine/L5-Swagger](https://github.com/DarkaOnLine/L5-Swagger) to see if I can do it for the project, but it doesn't seem to be a one-step thing. It's giving me errors. I'm not a PHP dev, so most of the errors are Greek to me and could take me longer to figure out.

**Comment by asbiin** [comment]
Actually I tried with laravel-apidoc-generator, you can see here #2552, but there is still a lot of work.

**Comment by Mohammed90** [comment]
Yeah, after resolving the errors, I realized it's not just a matter of including the dependency and pointing it at the `api/` directory. The classes/methods still need to be annotated.
